cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -371,6 +412,30 @@ private void getAppDetails(HttpResponder responder, String namespace, String nam
     }
   }
 
+  // parse a filename into an Id.Artifact, assuming the filename is of format {name}-{version}.jar
```

Comment:
yeah i will need to add that logic, but we should try and get from manifest if the header is not present. Opened CDAP-3310